src/main/kotlin/kweb/Kweb.kt

```diff
@@ -169,45 +169,39 @@ class Kweb private constructor(
+                ?: error("Client id $sessionId not found")
         wsClientData.lastModified = Instant.now()
         val debugToken: String? = if(!debug) null else {
             val dt = abs(random.nextLong()).toString(16)
             wsClientData.debugTokens[dt] = DebugInfo(javascript, "executing", Throwable())
             dt
         }
-        server2ClientMessage.debugToken = debugToken
+        funcCall.debugToken = debugToken
         val outboundMessageCatcher = outboundMessageCatcher.get()
         if (outboundMessageCatcher == null) {
-            wsClientData.send(server2ClientMessage)
+            wsClientData.send(Server2ClientMessage(sessionId, funcCall))
         } else {
-            logger.debug("Temporarily storing message for ${server2ClientMessage.yourId} in threadlocal outboundMessageCatcher")
-            val jsFunction = JsFunction(server2ClientMessage.jsId!!, server2ClientMessage.arguments!!)
+            logger.debug("Temporarily storing message for $sessionId in threadlocal outboundMessageCatcher")
+            val jsFunction = JsFunction(funcCall.jsId!!, funcCall.arguments)
             outboundMessageCatcher.add(jsFunction)
             //Setting `shouldExecute` to false tells the server not to add this jsFunction to the client's cache,
             //but to not actually run the code. This is used to pre-cache functions on initial page render.
-            server2ClientMessage.shouldExecute = false
-            wsClientData.send(server2ClientMessage)
+            funcCall.shouldExecute = false
+            wsClientData.send(Server2ClientMessage(sessionId, funcCall))
         }
     }
 
-    fun callJsWithCallback(server2ClientMessage: Server2ClientMessage,
+    fun callJsWithCallback(sessionId: String, funcCall: FunctionCall,
                            javascript: String, callback: (JsonElement) -> Unit) {
         //TODO I could use some help improving this error message
-        require(outboundMessageCatcher.get() == null) { "Can not use callback while page is rendering" }
-        val wsClientData = clientState.getIfPresent(server2ClientMessage.yourId)
-                ?: error("Client id ${server2ClientMessage.yourId} not found")
+        //require(outboundMessageCatcher.get() == null) { "Can not use callback while page is rendering" }
+        val wsClientData = clientState.getIfPresent(sessionId)
+                ?: error("Client id $sessionId not found")
         wsClientData.lastModified = Instant.now()
-        val debugToken: String? = if(!debug) null else {
-            val dt = abs(random.nextLong()).toString(16)
-            wsClientData.debugTokens[dt] = DebugInfo(javascript, "executing", Throwable())
-            dt
-        }
-        server2ClientMessage.debugToken = debugToken
-        wsClientData.handlers[server2ClientMessage.callbackId!!] = callback
-        wsClientData.send(server2ClientMessage)
+        wsClientData.handlers[funcCall.callbackId!!] = callback
```

Comment:
Might want to use `:? error(...)` to provide a more useful error message if callbackId is null.